pkg/asset/manifests/dns.go

```diff
@@ -58,6 +71,23 @@ func (d *DNS) Generate(dependencies asset.Parents) error {
 		},
 	}
 
+	switch installConfig.Config.Platform.Name() {
+	case awstypes.Name:
+		zone, err := icaws.GetPublicZone(installConfig.Config.BaseDomain)
+		if err != nil {
+			return errors.Wrapf(err, "getting public zone for %q", installConfig.Config.BaseDomain)
+		}
+		config.Spec.PublicZone = &configv1.DNSZone{ID: strings.TrimPrefix(*zone.Id, "/hostedzone/")}
+		config.Spec.PrivateZone = &configv1.DNSZone{Tags: map[string]string{
+			"openshiftClusterID":                                                          clusterID.ClusterID,
+			fmt.Sprintf("kubernetes.io/cluster/%s", installConfig.Config.ObjectMeta.Name): "owned",
+			"Name": fmt.Sprintf("%s_int", installConfig.Config.ObjectMeta.Name),
+		}}
+	case libvirttypes.Name, openstacktypes.Name, nonetypes.Name:
+	default:
+		return errors.New("invalid Platform")
```

Comment:
Isn't our current plan to ignore invalid platforms in most places?  See [here][1] and [here][2].  I guess your current whitelist approach makes sense if we expect new platforms to require cluster-managed DNS configuration and that we feel a more permissive approach in this block would make it easier to forget to add it?

[1]: https://github.com/openshift/installer/pull/792#discussion_r253683216
[2]: https://github.com/openshift/installer/pull/1112/files#diff-6532666297c6115f7774f93ebab396c4R156